met-etl/invoke_jobs.py

```
@@ -0,0 +1,75 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generate account statements.
+
+This module will create statement records for each account.
+"""
+import os
+import sys
+
+from flask import Flask
+from utils.logger import setup_logging
+
+import config
+
+
+
+setup_logging(os.path.join(os.path.abspath(os.path.dirname(__file__)), 'logging.conf'))  # important to do this first
+
+
+def create_app(run_mode=os.getenv('FLASK_ENV', 'production')):
+    """Return a configured Flask App using the Factory method."""
+    from  met_etl.models import db, ma
+
+    app = Flask(__name__)
+
+    app.config.from_object(config.CONFIGURATION[run_mode])
+    # Configure Sentry
+    app.logger.info(f'<<<< Starting ETL Jobs >>>>')
+    db.init_app(app)
+    ma.init_app(app)
+
+    register_shellcontext(app)
+
+    return app
+
+
+def register_shellcontext(app):
+    """Register shell context objects."""
+
+    def shell_context():
+        """Shell context objects."""
+        return {
+            'app': app
+        }  # pragma: no cover
+
+    app.shell_context_processor(shell_context)
+
+
+def run(job_name):
+    from tasks.met_extractor import MetExtractor
+    application = create_app()
+
+    application.app_context().push()
+    if job_name == 'EXTRACT_MET':
+        MetExtractor.do_etl()
+        application.logger.info(f'<<<< Completed MET Extraction >>>>')
+    else:
+        MetExtractor.do_etl()
```

make it error out instead of having a default